### ISSUE TITLE
We shouldn't let people run incomplete templated SQL queries 

### ISSUE THREAD

**Issue by camsaul** [metabase/metabase#10928]
If you add a new variable but don't add info like `Variable Type` the Query Processor returns an error, because `:type` is required for parameters. We shouldn't let someone run an invalid query in the first place – we should make sure they pick things like type first before running.

The client is attempting to run a query that looks something like:

```clj
{:native
 {:query         "SELECT *\nFROM ORDERS\nWHERE TOTAL > 100 [[ AND {{created}} # ]] CREATED_AT < now()",
  :template-tags {:created {:id           "0ff12c0a-92d2-1f02-0527-a680928df563"
                            :name         "created"
                            :display_name "Created"
                            :type         nil}}}}
```

![Screen Shot 2019-09-18 at 1 08 36 PM](https://user-images.githubusercontent.com/1455846/65182136-77c3cd80-da15-11e9-88ce-b99fe64cf0bd.png)

The error message is almost useless so I will open a separate issue for that as well.

**Comment by mazameli** [comment]
So I made a dumb choice eons ago when we first implemented SQL variables, which was to not have their be a default selection for `Variable type`. The simple way to solve this is to just have `Variable type` default to `Text`.

**Comment by paulrosenzweig** [comment]
Fixed by #10946
